content/en/blog/2019/monitoring-external-service-traffic/index.md

```diff
@@ -0,0 +1,360 @@
+for these cases. This blog will cover how you can use the augmented metrics to
+monitor all external service traffic.
+
+The Istio control plane configures the sidecar proxy with
+predefined clusters called `BlackHoleCluster` and `Passthrough` which block or
+allow all traffic respectively. Let's start with understanding what `BlackHole`
+and `Passthrough` clusters mean in the context of Istio networking.
+
+### What are `BlackHole` and `Passthrough` clusters?
+
+Istio provides users the option to configure how traffic for external services
+is handled by default in the mesh. The Istio control plane by default
+creates and provides configuration for all Kubernetes services to all the sidecars
+in order to preserve the default Kubernetes behavior that all services within the cluster
+are always reachable. This means that unless Istio RBAC or Kubernetes network policies
+are configured, all Kubernetes services continue to communicate with each other seamlessly
+after Istio is added. For services outside the Kubernetes cluster i.e. external
+services, Istio provides two options, first to block all external service access
+(enabled  by setting `global.outboundTrafficPolicy.mode` to `REGISTRY_ONLY`) and
+second to allow all access to external service (enabled  by setting
+`global.outboundTrafficPolicy.mode` to `ALLOW_ANY`). The default option for this
+setting is to allow all external service access. This is where the `BlackHole`
+and `Passthrough` clusters are used.
+
+* **`BlackHoleCluster`** - The `BlackHoleCluster` is a fake virtual cluster created
+  in the Envoy configuration when `global.outboundTrafficPolicy.mode` is set to
+  `REGISTRY_ONLY`. In this mode, all traffic to external service is blocked unless
+  [`ServiceEntries`](/docs/reference/config/networking/v1alpha3/service-entry)
+  are explicitly added for each service. To implement this, the default virtual
+  outbound listener at `0.0.0.0:15001` which uses `SO_ORIGINAL_DST` is setup
+  as a TCP Proxy with the `BlackHoleCluster` as the static cluster. The configuration
+  for the `BlackHoleCluster` looks like this:
+
+  {{< text json >}}
+    {
+      "name": "BlackHoleCluster",
+      "type": "STATIC",
+      "connectTimeout": "10s"
+    }
+  {{< /text >}}
+
+  As you can see, this cluster is static with no endpoints so all the traffic
+  will be dropped. Additionally, Istio creates unique listeners for every
+  port/protocol combination of platform services which gets hit instead of the
+  virtual listener if the request is made to an external service on the same port.
+  In that case, the route configuration of every virtual route in Envoy is augmented to
+  add the `BlackHoleCluster` like this:
+
+  {{< text json >}}
+    {
+      "name": "block_all",
+      "domains": [
+        "*"
+      ],
+      "routes": [
+        {
+          "match": {
+            "prefix": "/"
+          },
+          "directResponse": {
```

Comment:
is it worth explaining `directResponse` here (in contrast to `route` for the passthru)?

Reply:
done